src/store/Controller.cc

```diff
@@ -276,6 +276,10 @@ Store::Controller::dereferenceIdle(StoreEntry &e, bool wantsLocalMemory)
     if (EBIT_TEST(e.flags, ENTRY_SPECIAL))
         return true;
 
+    // idle private entries cannot be reused
```

Comment:
> AIUI this restriction is unrelated to being idle. So for clarity best to avoid that word here.

It _is_ related: Non-idle private entries are reused today. Being private is about discoverability by others. If two transactions managed to discover an entry _before_ it became private those transactions will continue to share it. See StoreEntry::shareableWhenPrivate description and commit 39fe14b.

> However, what about the StoreEntry::shareableWhenPrivate flag ? this will delete shareable/reusable objects when it is true.

A true StoreEntry::shareableWhenPrivate flag helps transactions that already have a lock on the entry. The entry in question is idle (i.e. has no locks/users/transactions). The idle entry should be deleted even if that flag is true -- nobody can start reusing it because nobody can find it (and anybody who has found it earlier has abandoned/unlocked it by now).

Please withdraw your change request.

Reply:
Aha. Thanks for the explanation.